### DISCUSSION TITLE
Integration of Ethernet LAN8720 Board with ESP32

### DISCUSSION THREAD

**Discussion by tefod-zz** [arendst/Tasmota#20096] [Q&A]
Hi Community,
I am working on integrating an Ethernet LAN8720 board with an ESP32 and am looking to configure it within Tasmota. 
Despite extensive searches online, I have not found any detailed examples or guides on how to wire the LAN8720 board to the ESP32 and then configure it in Tasmota?

**Device**: [ESP32 38pin cp2102 ](https://www.aliexpress.com/item/1005005940686609.html)
**Additional Hardware**: [Ethernet LAN8720 Board](https://www.aliexpress.com/item/1005006005000260.html)

- What is the correct wiring schematic for connecting the LAN8720 board to the ESP32?
- Has anyone in the community successfully implemented this setup and, if so, could you share your insights or resources?

Additionally, I would like to mention that for this project, I plan to use and compile the "ethernet" variant of the Tasmota firmware myself. I am also familiar with the [WT32-ETH01 module](https://www.aliexpress.com/item/1005005967733312.html) and have used it in other projects. However, my goal here is to build and configure the setup myself with the LAN8720 board for a better understanding and customization.

Thank you in advance for your assistance :)

![20231124_130213](https://github.com/arendst/Tasmota/assets/3308645/d605f582-a27d-4524-a7d8-0fbba6710bf4)
![20231124_130231](https://github.com/arendst/Tasmota/assets/3308645/17dfd2c7-171c-4825-97a3-d395a3647e6a)

**Comment by sfromis** [comment]
If that separate card can work, it should be connected to regular ESP32 Ethernet pins also used by WT32-ETH01. I've not got experience making such setups work, but I think that the pins listed here (very much including the dedicated pins) apply:
https://github.com/arendst/Tasmota/blob/development/tasmota/tasmota_xdrv_driver/xdrv_82_esp32_ethernet.ino#L23-L58

**Comment by Jason2866** [comment]
If you really want to do ;-) People had success following this guide https://sautter.com/blog/ethernet-on-esp32-using-lan8720/
Some tips here https://github.com/espressif/arduino-esp32/issues/2907
